src/traces/scattergeo/plot.js

```diff
@@ -72,66 +66,39 @@ module.exports = function plot(geo, calcData) {
         }
 
         if(subTypes.hasMarkers(trace)) {
-            s.selectAll('path.point').data(_calcTrace)
-              .enter().append('path')
-                .classed('point', true);
+            s.selectAll('path.point')
+                .data(Lib.identity)
+             .enter().append('path')
+                .classed('point', true)
+                .each(function(calcPt) { removeBADNUM(calcPt, this); });
```

Comment:
For the record, I just had a discussion with @etpinard about this where my gut reaction was that we should do
```js
s.selectAll('path.point')
    .data(filterOutBadPoints)
  .enter().append('path')
    ...
```
instead of creating all the nodes and then deleting the bad ones.

The conclusion we came to though is that the way it is here is actually better, despite being a bit less "d3-idiomatic." The rationale is that bad points should be considered an edge case, so we don't worry about the overhead of creating and deleting their nodes (which will happen on every replot), just about how the `.each` call to test (good) points for removal compares to whatever `filterOutBadPoints` does. And since the latter would *always* need to duplicate the `calcdata` array, it's almost certainly slower than this `.each`.